test/common.js

```diff
@@ -72,109 +84,171 @@ SessionOutput.prototype._unqueueWait = function _unqueueWait() {
     this.waitQueue.shift()();
 };
 
-SessionOutput.prototype.wait = function wait(regexp, callback) {
-  if (!this._queueWait(() => { this.wait(regexp, callback); }))
+SessionOutput.prototype.timeoutAfter = function timeoutAfter(timeout) {
+  this.timeout = timeout;
+};
+
+SessionOutput.prototype.wait = function wait(regexp, callback, allLines) {
+  if (!this._queueWait(() => { this.wait(regexp, callback, allLines); }))
     return;
 
   const self = this;
-  this.on('line', function onLine(line) {
+  const lines = [];
+
+  function onLine(line) {
+    lines.push(line);
+    debug('[LINE]', line);
+
     if (!regexp.test(line))
       return;
 
     self.removeListener('line', onLine);
     self._unqueueWait();
+    done = true;
 
-    callback(line);
-  });
-};
-
-SessionOutput.prototype.waitBreak = function waitBreak(callback) {
-  this.wait(/Process \d+ stopped/i, callback);
-};
-
-SessionOutput.prototype.linesUntil = function linesUntil(regexp, callback) {
-  if (!this._queueWait(() => { this.linesUntil(regexp, callback); }))
-    return;
-
-  const lines = [];
-  const self = this;
-  this.on('line', function onLine(line) {
-    lines.push(line);
+    callback(null, allLines ? lines : line);
+  }
 
-    if (!regexp.test(line))
+  let done = false;
+  const check = setTimeout(() => {
+    if (done)
       return;
 
     self.removeListener('line', onLine);
     self._unqueueWait();
+    console.error(`${'='.repeat(10)} lldb output ${'='.repeat(10)}`);
```

Comment:
Are these `console.error()`s intentional?

Reply:
@cjihrig Yes, the intent is to log the lldb output that `wait` fails to match the expected regexp against, even with TEST_LLNODE_DEBUG turned off. After this log the error will be passed to the callback and the test should fail immediately.